### PR TITLE
symbolic paintable

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: gtk4/src/subclass/symbolic_paintable.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::subclass::prelude::*;
+use crate::SymbolicPaintable;
+use glib::translate::*;
+use glib::Cast;
+
+pub trait SymbolicPaintableImpl: ObjectImpl {
```
Comment (sdroege) [inline gtk4/src/subclass/symbolic_paintable.rs:8]: Doesn't this required Paintable?

File: gtk4/src/subclass/symbolic_paintable.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    let imp = instance.impl_();
+
+    let snapshot: Borrowed<gdk::Snapshot> = from_glib_borrow(snapshotptr);
+
+    imp.snapshot_symbolic(
+        from_glib_borrow::<_, SymbolicPaintable>(paintable).unsafe_cast_ref(),
+        &snapshot,
+        width,
+        height,
+        FromGlibContainer::from_glib_none_num(colors, n_colors),
```
Comment (sdroege) [inline gtk4/src/subclass/symbolic_paintable.rs:95]: Should probably just pass this on as &[
gdk::RGBA]. Just go via slice::from_raw_parts()

File: gtk4/src/symbolic_paintable.rs [hunk 3]
```diff
@@ -8,7 +8,7 @@ pub trait SymbolicPaintableExtManual: 'static {
     #[doc(alias = "gtk_symbolic_paintable_snapshot_symbolic")]
     fn snapshot_symbolic(
         &self,
-        snapshot: &impl IsA<gdk::Snapshot>,
+        snapshot: &gdk::Snapshot,
```
Comment (sdroege) [inline gtk4/src/symbolic_paintable.rs:11]: Why?

Comment (bilelmoussaoui) [reply gtk4/src/symbolic_paintable.rs:11]: We don't do that anywhere else; ideally we would just cast and make the api uses gtk::Snapshot here really

File: gtk4/src/subclass/symbolic_paintable.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply gtk4/src/subclass/symbolic_paintable.rs:8]: It does indeed

File: gtk4/src/subclass/symbolic_paintable.rs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+use glib::Cast;
+
+pub trait SymbolicPaintableImpl: PaintableImpl {
+    fn snapshot_symbolic(
+        &self,
+        paintable: &Self::Type,
+        snapshot: &gdk::Snapshot,
+        width: f64,
+        height: f64,
+        colors: [gdk::RGBA],
```
Comment (sdroege) [inline gtk4/src/subclass/symbolic_paintable.rs:16]:
```suggestion
        colors: &[gdk::RGBA],
```

**Review by sdroege — APPROVED** [review]

**Pull request merged**